src/state/status/sagas.ts

```diff
@@ -0,0 +1,68 @@
+import NetInfo from '@react-native-community/netinfo';
+import { eventChannel } from 'redux-saga';
+import { takeLatest, take, put, call, delay } from 'redux-saga/effects';
+
+import { addToastMessage } from '../toastMessages/actions';
+import {
+  UpdateServerConnectionStatusAction,
+  UpdateInternetConnectionStatusAction,
+  updateInternetConnectionStatusSuccess,
+  StatusAction,
+  updateServerConnectionStatusSuccess,
+} from './actions';
+
+const BlueElectrum = require('../../../BlueElectrum');
+const i18n = require('../../../loc');
+
+function emitNetInfoStatus() {
+  return eventChannel(emitter => {
+    const eventListener = NetInfo.addEventListener(state => {
+      emitter(state);
+    });
+
+    return () => {
+      eventListener();
+    };
+  });
+}
+
+export function* updateServerConnectionStatusSaga(action: UpdateServerConnectionStatusAction | unknown) {
+  while (true) {
+    yield delay(10000);
```

Comment:
not sure if that's the best way to check server status

@KedziaPawel do you have any ideas?

Reply:
you should subscribe to events in electrum-client like onClose, onConnect, reconnect etc.